addons/stock/report/report_stock_forecast.py

```diff
@@ -91,25 +96,59 @@ def init(self):
+     FROM
+     (
+            SELECT date_trunc('day', CURRENT_DATE) AS DATE
+            UNION ALL
+            SELECT date(sm.date_expected) AS date
+            FROM stock_move sm
+            LEFT JOIN
+            stock_location source_location ON sm.location_id = source_location.id
+            LEFT JOIN
+            stock_location dest_location ON sm.location_dest_id = dest_location.id
+            WHERE
+            sm.state IN ('confirmed','assigned','waiting') and sm.date_expected > CURRENT_DATE and
+            ((dest_location.usage = 'internal' AND source_location.usage != 'internal')
+             or (source_location.usage = 'internal' AND dest_location.usage != 'internal'))) AS DATE_SEARCH)
+            SUB ON (SUB.date IS NOT NULL)
+    GROUP BY MAIN.product_id,SUB.date, MAIN.date, MAIN.company_id,MAIN.reference
     ) AS FINAL
-    GROUP BY product_id,date,company_id)""")
+    LEFT JOIN
+        stock_picking sp on final.reference=sp.name
+    Where final.product_qty != 0 or final.reference = 'Starting Inventory'
+    GROUP BY product_id,final.date,final.id,final.company_id,final.reference,sp.priority,sp.date,sp.id)""")
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
+        orderby = 'date, id' if not orderby else orderby  # Sort result by date_expected and id
+        fields.append('quantity') if 'quantity' not in fields else fields
+        res = super(ReportStockForecat, self).read_group(domain, fields, groupby, offset=offset, limit=limit, orderby=orderby, lazy=lazy)
+        lists = []
+        final_val = {'qty_for_final_line': [], 'cumulative_qty_final_line': []}
+        res_list = {}
+        if 'cumulative_quantity' in fields:
+            for line in res:
+                if line.get('product_id'):
+                    index = res.index(line)
+                    if not res_list.get(line['product_id']):
+                        res_list.update({line['product_id']: []})
+                    res_list[line['product_id']].append(line['quantity'])  # Get cumulative quantity product wise
+                    line['cumulative_quantity'] = sum(res_list[line['product_id']])  # Sum of all quantities (i.e. cumulative quantity) product wise
+                    if line.get('reference') == 'Starting Inventory':
+                        line['quantity'] = line['cumulative_quantity'] if line.get('__count') == 1 else line['quantity']   # Set quantity if line's quantity is 0.0
+                        if index > 1 and line.get('__count') > 1:
+                            line['cumulative_quantity'] = res[index-1]['cumulative_quantity']  # Set cumulatiove quantity if line does not have any cumulative
+                else:
+                    lists.append(line['quantity']) if line.get('quantity') else lists
+                    line['cumulative_quantity'] = sum(lists)  # Sum of all Move Operations' quantity
+                if not line.get('reference'):
```

Comment:
@hch-odoo are you sure the part after this line is useful? I could translate it by if not groupby and 'reference' not in groupby then line['quantity'] = line['quantity']